docs/checkit.js

```diff
@@ -782,59 +782,30 @@ CheckIt.prototype.findCurrentDay = function() {
 };
 
 
-// Should these be CheckIt functions? or Calendar. TODO
-// How to access the calendar object when I need it for these functions?
-// current active calendar...??
+// calendarAnalyzer takes a calendar object and can analyze facts about it.
 
-CheckIt.prototype.getNumberOfChecked = function() {
-    // Returns the number of checked days in a calendar
-
-    var checkit = this;
+var CalendarAnalyzer = function(calendarState) {
+    this.calState = calendarState;
+};
 
-    return checkit.store.getActive().then(function(activeCalendarId) {
-        
-        
-        
-        
-        /*console.log("found this activeCalendarId: " + activeCalendarId);
-    return checkit.store.getActive()
-        .then(function(activeCalendarId) {
-            return checkit.store.loadById(activeCalendarId)
-        .catch(function(err) {
-            console.log("There is no current active calendar " + err);
-            checkit.store.removeActive();
-            checkit.showForm(checkit.$buildCalendarForm);
-            return err;
-            });
-        })
-        .then(function(activeCalendarState) {
-            if (activeCalendarState !== null) {
-                var state = activeCalendarState;
-                var calendar = new Calendar(state);
-                return checkit.displayCalendar(calendar);
-            }
-        })
-        .catch(function(err) {
-            console.log("Could not display active calendar" + err);
-            checkit.showForm(checkit.$buildCalendarForm);
-            return err;
-        });
-            * 
-            * 
-            */
-            
-            
-            
-        });
+CalendarAnalyzer.prototype.getNumberOfChecked = function() {
+    // Returns the number of checked days in a calendar
     
+    var size = 0, key;
+    for (key in this.calState.checkedDays) {
```

Comment:
I think you can declare variables in the loop directly, although it doesn't really change anything semantically.

`for (var key in this...)`